Content.Shared/GameObjects/Components/Mobs/State/SharedMobStateComponent.cs

```diff
@@ -0,0 +1,334 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Content.Shared.Alert;
+using Content.Shared.GameObjects.Components.Damage;
+using Content.Shared.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Shared.GameObjects.Components.Mobs.State
+{
+    /// <summary>
+    ///     When attacked to an <see cref="IDamageableComponent"/>,
+    ///     this component will handle critical and death behaviors for mobs.
+    ///     Additionally, it handles sending effects to clients
+    ///     (such as blur effect for unconsciousness) and managing the health HUD.
+    /// </summary>
+    public abstract class SharedMobStateComponent : Component, IMobStateComponent, IActionBlocker
+    {
+        public override string Name => "MobState";
+
+        public override uint? NetID => ContentNetIDs.MOB_STATE;
+
+        /// <summary>
+        ///     Thresholds for this <see cref="SharedMobStateComponent"/>,
+        ///     mapped from threshold to its corresponding state.
```

Comment:
Maybe explain what a threshold is (since it's just an int)

Reply:
Done.